tests/test_tensorboard.py

```diff
@@ -18,21 +21,19 @@
 
 
 class HParamCallback(BaseCallback):
-    def __init__(self):
-        """
-        Saves the hyperparameters and metrics at the start of the training, and logs them to TensorBoard.
-        """
-        super().__init__()
+    """
+    Saves the hyperparameters and metrics at the start of the training, and logs them to TensorBoard.
+    """
 
     def _on_training_start(self) -> None:
-        hparam_dict = {
-            "algorithm": self.model.__class__.__name__,
-            "learning rate": self.model.learning_rate,
-            "gamma": self.model.gamma,
-        }
+        hparam_dict: Dict[str, Union[str, float]] = {"algorithm": self.model.__class__.__name__}
+        if isinstance(self.model.learning_rate, float):  # Can also be Schedule, in that case, we don't report
+            hparam_dict["learning rate"] = self.model.learning_rate
+        if isinstance(self.model, (OnPolicyAlgorithm, OffPolicyAlgorithm)):
+            hparam_dict["gamma"] = self.model.gamma
```

Comment:
Currently, gamma is defined in `OnPolicyAlgorithm` and `OffPolicyAlgorithm`, but not in `BaseAlgorithm`. I am not very satisfied with this solution, but it works.

Reply:
gamma makes sense for most algorithms but not ES ones like ARS/CEM (actually we could set gamma=1 for them for consistency even though it is not used)

Reply:
> I am not very satisfied with this solution, but it works.

yes, not pretty but:
1. it is for a test
2. you could probably ignore mypy here